src/consensus/params.h

```diff
@@ -95,6 +95,23 @@ struct Params {
 
     bool fSimplifiedRewards;
     bool fShortEarlyCoinbase;
+
+    /** Auxpow parameters */
+    int32_t nAuxpowChainId;
+    bool fStrictChainId;
+    int nLegacyBlocksBefore; // -1 for "always allow"
```

Comment:
The problem I have with this attribute name and logic further down is that "legacy" is very ambiguous: From a 1.8.x point of view, **legacy** blocks are anything that does not have the chainId and version == 2, i.e. must be `version & 0x006200ff  == 0x00620002` or is legacy. However, from a 1.10 perspective, the condition becomes `0x00620003` for a block to not be legacy, and for 1.14, anything that doesn't match `0x00620004` is legacy... so this simply doesn't work well as a reference descriptor.

Additionally, other attributes that specify a height in `Params` are **enabling** a feature whereas this seems to **disable** a feature, but that is not what happened. AuxPoW enabled new blocks, but did not disable things. In fact, 1.8.0 was a hard fork that explicitly enabled a new functionality at that specific height, so this is imho the worst possible place to deviate.

Reply:
So it's got that name because we used a similar term in the previous release ( https://github.com/dogecoin/dogecoin/blob/master/src/chainparams.cpp#L121 ). So we'd be diverging from both our release and Namecoin if we change it. Beyond that, sure, happy to, but wanted to raise that.

Reply:
The boolean you reference is ambiguous too, but since it already changed into an int and is now your proposed primary activator for AuxPoW instead of a tree structure, so I would say that we're already diverging from our previous release (and I tbh don't think we should care much about diverging from Namecoin's variable naming)

I am okay with divergence in any case, if it makes things easier to understand. This, in my opinion, makes things harder to understand.